debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/pgoutput/PgOutputMessageDecoder.java

```diff
@@ -0,0 +1,645 @@
+        if (LOGGER.isTraceEnabled()) {
+            LOGGER.trace("Event: {}", MessageType.COMMIT);
+            LOGGER.trace("Flags: {} (currently unused and most likely 0)", flags);
+            LOGGER.trace("Commit LSN: {}", lsn);
+            LOGGER.trace("End LSN of transaction: {}", endLsn);
+            LOGGER.trace("Commit timestamp of transaction: {}", commitTimestamp);
+        }
+    }
+
+    /**
+     * Callback handler for the 'R' relation replication message.
+     *
+     * @param buffer The replication stream buffer
+     * @param typeRegistry The postgres type registry
+     */
+    private void handleRelationMessage(ByteBuffer buffer, TypeRegistry typeRegistry) throws SQLException {
+        int relationId = buffer.getInt();
+        String namespace = readString(buffer);
+        String relationName = readString(buffer);
+        int replicaIdentityId = buffer.get();
+        short columnCount = buffer.getShort();
+
+        if (LOGGER.isTraceEnabled()) {
+            LOGGER.trace("Event: {}, RelationId: {}, Replica Identity: {}, Columns: {}", MessageType.RELATION, relationId, replicaIdentityId, columnCount);
+            LOGGER.trace("Schema: '{}', Table: '{}'", namespace, relationName);
+        }
+
+        List<ColumnMetaData> columns = new ArrayList<>();
+        for (short i = 0; i < columnCount; ++i) {
+            byte flags = buffer.get();
+            String columnName = Strings.unquoteIdentifierPart(readString(buffer));
+            int columnType = buffer.getInt();
+            int attypmod = buffer.getInt();
+
+            final PostgresType postgresType = typeRegistry.get(columnType);
+            boolean key = (flags & 1) == 1;
+
+            columns.add(new ColumnMetaData(columnName, postgresType, key, attypmod));
+        }
+
+        PgOutputRelationMetaData metadata = new PgOutputRelationMetaData(relationId, namespace, relationName, columns);
+        if (relationIdMetadata.containsKey(relationId)) {
+            LOGGER.warn("Prior relation message for '{}' has not been consumed by a DML; being replaced.", relationId);
+        }
+
+        TableId tableId = new TableId(null, namespace, relationName);
+
+        if (LOGGER.isTraceEnabled()) {
+            LOGGER.trace("Cached relation '{}' with table-id '{}' and synchronizing schema", relationId, tableId);
+        }
+
+        relationIdMetadata.put(relationId, metadata);
+        tableIdByRelationId.put(relationId, tableId);
+
+        synchronizeSchema(relationId, metadata);
+    }
+
+    private void synchronizeSchema(Integer relationId, PgOutputRelationMetaData metadata) throws SQLException {
+        // todo: DBZ-766 a bit of this is based on RecordsStreamProducer#columnValues
+        //      Can we avoid some of the code dupe somehow?
+        if (metadata.getColumns() == null || metadata.getColumns().isEmpty()) {
+            LOGGER.warn("No columns detected, nothing to synchronize");
```

Comment:
In hindsight, I do not believe its technically possible to ever find ourselves in a scenario where the metadata would have 0 columns since the logical replication requires that there be a primary key column defined for tables to have changes published.  So at a minimum, the metadata would always have columns that represent the replica identity, whether that is just the primary key columns with replica identity default or all columns with replica identity full.

That said, I'm fairly certain this change is superfluous.